starter-code/pizza.js

```diff
@@ -45,26 +45,120 @@ function renderPepperonni() {
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(function($mushroom){
+    if (state.mushrooms) {
+      $mushroom.style.visibility = "visible";
+    }
+    else {
+      $mushroom.style.visibility = "hidden";
+    }
+  })
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(function($green){
+    if (state.greenPeppers) {
+      $green.style.visibility = "visible";
+    }
+    else {
+      $green.style.visibility = "hidden";
+    }
+  })
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll('.sauce').forEach(function($sauce){
+    if (state.whiteSauce) {
+      $sauce.style.visibility = "visible";
+    }
+    else {
```

Comment:
normally if you have no white sauce, you should see a red tomato sauce. For that, just write it like you did for the crust.